crates/devices/virtio-vsock/src/packet.rs

```diff
@@ -460,22 +461,34 @@ impl<'a, B: BitmapSlice> VsockPacket<'a, B> {
             return Err(Error::InvalidHeaderLen(pkt.len()));
         }
 
-        let data_desc = desc_chain.next().ok_or(Error::DescriptorChainTooShort)?;
+        // Starting from Linux 6.2 the virtio-vsock driver can use a single descriptor for both
+        // header and data.
+        let data_slice =
+            if !chain_head.has_next() && chain_head.len() - PKT_HEADER_SIZE as u32 >= pkt.len() {
+                mem.get_slice(
+                    chain_head
+                        .addr()
+                        .checked_add(PKT_HEADER_SIZE as u64)
+                        .ok_or(Error::DescriptorLengthTooSmall)?,
+                    pkt.len() as usize,
+                )
+                .map_err(Error::InvalidMemoryAccess)?
+            } else {
+                let data_desc = desc_chain.next().ok_or(Error::DescriptorChainTooShort)?;
```

Comment:
nit: I would check before is the next bit of the head is set (with has_next()).

Reply:
Okay, I think it was pre-existing since I just moved the code in the else branch, but since I'm touching this code, make sense to update it ;-)